src/core/tranche/seniorOperator.sol

```diff
@@ -41,44 +29,27 @@ contract Quant is DSNote {
     
     Rate public borrowRate;
     uint public debt;
-    uint public supplyRate;
-    bool public supplyRateFixed;
-    
-    constructor(address reserve_) public {
-        reserve = ReserveLike(reserve_);
-        wards[msg.sender] = 1;
+
+    constructor(address reserve_, address slicer_) Operator(reserve_, slicer_) public {
         borrowRate.index = ONE;
         borrowRate.speed = ONE;
-        supplyRate = ONE;
     }
 
     function file(bytes32 what, uint speed_) public note auth {
          if (what ==  "borrowrate") {
             drip();
             borrowRate.speed = speed_;
         }
-        else if (what ==  "supplyrate") {
-            drip();
-            supplyRate = speed_;
-        }
     }
 
-    function file(bytes32 what, bool data) public note auth {
-         if (what ==  "fixedsupplyrate") {
-             supplyRateFixed = data;
-        }
+    function repay(address usr, uint currencyAmount) public note auth {
+        super.repay(usr, currencyAmount);
+        updateDebt(int(currencyAmount) * -1);
```

Comment:
shouldn't this be `int(currencyAmount * -1)`?

Reply:
also maybe we should updateDebt before repay (reentrancy)

Reply:
I think if I cast the currencyAmount to int then the operation is already performed in the int domain. But in general we could split it in increase and decrease debt to avoid type casts....